From: Ben Hutchings <ben@decadent.org.uk>
Date:
Subject: x86/apic: Avoid ABI change in 3.16.60
Forwarded: not-needed

Commit 288e4abffd07 "x86/apic: Fix signedness bug in APIC ID validity
checks" changed the type of apic::apic_id_valid, changing a parameter
type from int to u32.  This is ABI compatible, so hide it from
genksyms.

---
--- a/arch/x86/include/asm/apic.h
+++ b/arch/x86/include/asm/apic.h
@@ -288,7 +288,11 @@ struct apic {
 
 	int (*probe)(void);
 	int (*acpi_madt_oem_check)(char *oem_id, char *oem_table_id);
+#ifndef __GENKSYMS__
 	int (*apic_id_valid)(u32 apicid);
+#else
+	int (*apic_id_valid)(int apicid);
+#endif
 	int (*apic_id_registered)(void);
 
 	u32 irq_delivery_mode;
